ui/app/src/components/CreateProjectDialog/CreateProjectDialog.tsx

```diff
@@ -0,0 +1,106 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ChangeEvent, Dispatch, DispatchWithoutAction, useState } from 'react';
+import { Dialog, DialogTitle, DialogContent, DialogActions, Button, TextField, IconButton } from '@mui/material';
+import CloseIcon from 'mdi-material-ui/Close';
+
+export interface CreateProjectDialogProps {
+  open: boolean;
+  onClose: DispatchWithoutAction;
+  onConfirm: Dispatch<string>;
+}
+
+/**
+ * Dialog used to create a project.
+ * TODO: See with experts if the validation can't be done in a more straightforward way as it sounds complicate for
+ *   nothing. Just validating here a required field...
+ * @param props.open Define if the dialog should be opened or not.
+ * @param props.closeDialog Provides the function to close itself.
+ * @param props.onConfirm Action to perform when user confirmed.
+ * @constructor
+ */
+const CreateProjectDialog = (props: CreateProjectDialogProps) => {
+  const { open, onClose, onConfirm } = props;
+  const [projectValue, setProjectValue] = useState<string>('');
+  const [error, setError] = useState<string>('');
+
+  const triggerOnChange = (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
I think we have more of a convention of naming these types of functions in this way: `handleChange`, `handleConfirm`, `handleClose`, etc.

I'd update to that style